core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -127,7 +128,12 @@ private Optimizely(@Nonnull EventHandler eventHandler,
             if (getProjectConfig() != null) {
                 updateODPSettings();
             }
-            addUpdateConfigNotificationHandler(configNotification -> { updateODPSettings(); });
+            if (projectConfigManager != null) {
+                NotificationRegistry.getNotificationCenter(projectConfigManager.getSDKKey()).
```

Comment:
@jaeopt we are getting sdkKey from projectConfigManager instead of getProjectConfig() because of Asynchronous call of httpConfigManager it might not able to add this NotificationHandler when projectConfig is not fetched.